axelor-redmine/src/main/java/com/axelor/apps/redmine/service/imports/projects/RedmineProjectServiceImpl.java

```diff
@@ -104,32 +113,49 @@ public void redmineImportProject(
       TraceBackService.trace(e, "", batch.getId());
     }
 
-    // CREATE MAP FOR PASS TO THE METHODS
+    // CREATE POJO FOR PASS TO THE METHODS
 
-    HashMap<String, Object> paramsMap = new HashMap<String, Object>();
-
-    paramsMap.put("onError", onError);
-    paramsMap.put("onSuccess", onSuccess);
-    paramsMap.put("batch", batch);
-    paramsMap.put("errorObjList", errorObjList);
-    paramsMap.put("lastBatchUpdatedOn", lastBatchUpdatedOn);
-    paramsMap.put("redmineUserMap", redmineUserMap);
-    paramsMap.put("redmineProjectManager", redmineManager.getProjectManager());
+    MethodParameters methodParameters =
+        new MethodParameters(
+            onError,
+            onSuccess,
+            batch,
+            errorObjList,
+            lastBatchUpdatedOn,
+            redmineUserMap,
+            redmineManager);
 
     // IMPORT PROCESS
 
-    redmineImportProjectService.importProject(importProjectList, paramsMap);
-
-    // ATTACH ERROR LOG WITH BATCH
+    redmineImportProjectService.importProject(importProjectList, methodParameters);
+  }
 
-    if (errorObjList != null && errorObjList.size() > 0) {
-      MetaFile errorMetaFile = redmineErrorLogService.redmineErrorLogService(errorObjList);
+  private void fillUsersList(
+      RedmineManager redmineManager,
+      Map<Integer, Boolean> includedIdsMap,
+      List<User> redmineUserList,
+      Map<String, String> params)
+      throws RedmineException {
+    int offset = 0;
+    int limit = 25;
+    while (true) {
+      params.put("offset", String.valueOf(offset));
+      params.put("limit", String.valueOf(limit));
+
+      List<User> users = redmineManager.getUserManager().getUsers(params).getResults();
+
+      for (User user : users) {
+        if (!includedIdsMap.containsKey(user.getId())) {
+          redmineUserList.add(user);
+          includedIdsMap.put(user.getId(), true);
+        }
+      }
 
-      if (errorMetaFile != null) {
-        batch = batchRepo.find(batch.getId());
-        batch.setErrorLogFile(errorMetaFile);
-        batchRepo.save(batch);
+      if (users.size() < limit) {
```

Comment:
This should be the while's condition
If it doesn't work out of the box, use a 'do while' loop instead